scripts/processors/planner-pipette.lua

```diff
@@ -0,0 +1,189 @@
+local util = require("scripts.util")
+
+---@param index uint
+---@return LuaItemStack?
+local function get_cursor_stack(index)
+  local player = game.get_player(index) --[[@as LuaPlayer]]
+  local cursor = player.cursor_stack
+  if not (cursor and cursor.valid_for_read) then return end
+  return cursor
+end
+
+---@param event EventData.CustomInputEvent
+---@return LuaEntityPrototype?
+local function get_selected_prototype(event)
+  local selected = event.selected_prototype
+  if not selected then return end
+
+  if selected.base_type == "entity" then
+    local player = game.get_player(event.player_index) --[[@as LuaPlayer]]
+    local selected_entity = player.selected
+    if not selected_entity then
+      return game.entity_prototypes[selected.name]
+    end
+    if selected_entity.type == "entity-ghost" and selected_entity.ghost_prototype.type ~= "tile" then
+      return selected_entity.ghost_prototype --[[@as LuaEntityPrototype]]
+    end
+    return selected_entity.prototype
+
+  -- _CodeGreen: I couldn't find anything in game where I could click on an item or recipe that actually had a CustomInputEvent fire
```

Comment:
I think this is because of the `mouse-button-filter` on the GUI elements, which is unfortunate. Changing one of the keys to alt+leftclick makes it work, but that's already taken by recipe book and (in 2.0) Factoriopedia.

Reply:
yeah, I figured it had something to do with the choice of keybind, but I doubt most people will actually care, 99% of the time people are deconstructing or upgrading entities themselves, so I don't think it's that big of a deal. Regardless, if someone changes the keybind, the functionality is there at least, so there is that.